ui/components-react/src/components-react/filter-builder/FilterBuilder.tsx

```diff
@@ -0,0 +1,111 @@
+/*---------------------------------------------------------------------------------------------
```

Comment:
I tried to use `FilterBuilder` component and found a few usability issues.

Options to add a new rule or rule group only appear when hovering over a specific row of items. I often get lost when I need to add a new rule at the bottom of a large rule collection with nested rule groups. A better solution is to show these buttons at the group bottom whenever user hovers anywhere within the outline of this group.

When the `FilterBuilder` component is made fullscreen, scanning for the delete button becomes very uncomfortable. I think it's better to have it on the left-hand side.

Reply:
Made some changes to buttons placement (haven't pushed the changes yet): 
![image](https://user-images.githubusercontent.com/24278440/170065184-c1db4b10-45c4-403b-a4fc-746023ce8485.png)

All buttons are hidden until rule or group is hovered. @grigasp @roluk let me know what you think about these changes.

Reply:
I like this better. There needs to be some empty space between delete icon and textbox though.

Because the icons only appear  when use hovers on elements, this prevents all touch- and keyboard-only users from interacting with this UI. If we want to support these input methods, the basic solution would be to show all icons and buttons all the time, and the other option is to show the interactive elements when the related controls are in focus.

Reply:
I like this as well. 

> There needs to be some empty space between delete icon and textbox though.

Agree

> the basic solution would be to show all icons and buttons all the time, and the other option is to show the interactive elements when the related controls are in focus.

I think showing them all the time will make the UI "busy" again. I like the option to show them when related component is in focus.

Reply:
This is how it would look like without hiding buttons: 
![image](https://user-images.githubusercontent.com/24278440/170446213-bf1b74d2-ae86-4124-85ea-9d6efef9b497.png)

Pushed changes to show buttons when group or rule is focused instead of hovering. https://github.com/iTwin/itwinjs-core/pull/3655/commits/2efafbfaf9d83fe718cde0962be93a96dabd98f7

Reply:
Keyboard navigation works pretty well now, but hover behavior should not have been removed. Please also make only one set of delete and add rule/group buttons is visible at a time.

Let's also do something about enormous gaps between rule groups when no buttons are present. I'd like to try an approach where we are not reserving any space for the buttons until they are showed.